src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -281,6 +281,31 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
         public SyntaxNode ReturnTypeSyntax => CompilationUnit.Members.First(m => m.Kind() == SyntaxKind.GlobalStatement);
 
+        /// <summary>
+        /// Returns a bag of applied custom attributes and data decoded from well-known attributes. Returns null if there are no attributes applied on the symbol.
+        /// </summary>
+        /// <remarks>
+        /// Forces binding and decoding of attributes.
+        /// </remarks>
+        internal override CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
+        {
+            if (_lazyCustomAttributesBag == null || !_lazyCustomAttributesBag.IsSealed)
+            {
+                var mergedAttributes = ((SourceAssemblySymbol)this.ContainingAssembly).GetAttributeDeclarations();
```

Comment:
>GetAttributeDeclarations() [](http://example.com/codeflow?start=87&length=26)

Note that we are creating a ```SynthesizedSimpleProgramEntryPointSymbol``` per each compilation unit that has top-level statements. It doesn't feel right to bind all "main" attributes for every symbol like that. It looks like [https://github.com/dotnet/csharplang/issues/5045](https://github.com/dotnet/csharplang/issues/5045) suggests to allow "main" targeted attributes in any file, then perhaps we should bind attributes only for symbol that is returned by 
```
internal static SynthesizedSimpleProgramEntryPointSymbol? GetSimpleProgramEntryPoint(CSharpCompilation compilation)
```